File: solution/2500-2599/2530.Maximal Score After Applying K Operations/Solution.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        h = [-v for v in nums]
        heapify(h)
        ans = 0
        for _ in range(k):
            v = -heappop(h)
            ans += v
            heappush(h, -(ceil(v / 3)))
        return ans

File: solution/2500-2599/2530.Maximal Score After Applying K Operations/Solution.ts
function maxKelements(nums: number[], k: number): number {
    const pq = new MaxPriorityQueue<number>();
    nums.forEach(num => pq.enqueue(num));
    let ans = 0;
    while (k > 0) {
        const v = pq.dequeue();
        ans += v;
        pq.enqueue(Math.floor((v + 2) / 3));
        k--;
    }
    return ans;
}
